main/options.c

```diff
@@ -2363,6 +2365,18 @@ static void processMaxRecursionDepthOption (const char *const option, const char
 	Option.maxRecursionDepth = atol(parameter);
 }
 
+static void processPatternLengthLimit(const char *const option, const char *const parameter)
+{
+	if (parameter == NULL || parameter[0] == '\0')
+		error (FATAL, "A parameter is needed after \"%s\" option", option);
+
+	if (atol (parameter) < 0)
```

Comment:
Could you consider to use strtol?

I tried following small program:

```
[yamato@x201 tmp]$ cat foo.c
#include <stdlib.h>
int main(void) {return atoi("abc");} 
[yamato@x201 tmp]$ gcc foo.c
[yamato@x201 tmp]$ ./a.out ; echo $?
0
```

Atoi returns 0 when a unconvertable string is given.
I would like to reject such value here. A user giiving such value  is wrong. However, in some aspects, this limit is introduced to protect ctags from a kind of DoS attach. So I would like to deal with the option value carefully.